src/resolver.test.js

```diff
@@ -0,0 +1,169 @@
+const dns = require("dns");
+const {
+  Resolver,
+  InvalidRequestError,
+  ResolutionError,
+  NoSkynetDNSLinksFoundError,
+  MultipleSkylinksError,
+  InvalidSkylinkError,
+  MultipleSponsorKeyRecordsError,
+} = require("./resolver");
+
+jest.mock("dns");
+
+const FIXTURES = {
+  NO_SKYNET_LINKS: [["dnslink=/dummy-namespace/abcd-1234"]],
+  MULTIPLE_SKYNET_LINKS: [
+    ["dnslink=/skynet-ns/AQCYCPSmSMfmZjOKLX4zoYHHTNJQW2daVgZ2PTpkASFlSA"],
+    ["dnslink=/skynet-ns/AQCYCPSmSMfmZjOKLX4zoYHHTNJQW2daVgZ2PTpkASFlSB"],
+  ],
+  INVALID_SKYLINK: [["dnslink=/skynet-ns/broken-skylink"]],
+  MULTIPLE_SPONSOR_KEY: [
+    ["dnslink=/skynet-ns/AQCYCPSmSMfmZjOKLX4zoYHHTNJQW2daVgZ2PTpkASFlSA"],
+    ["skynet-sponsor-key=dummySponsorKey1"],
+    ["skynet-sponsor-key=dummySponsorKey2"],
+  ],
+  VALID_SPONSORED_SKYLINK: [
+    ["dnslink=/skynet-ns/AQCYCPSmSMfmZjOKLX4zoYHHTNJQW2daVgZ2PTpkASFlSA"],
+    ["skynet-sponsor-key=dummySponsorKey1"],
+  ],
+  VALID_SKYLINK: [["dnslink=/skynet-ns/AQCYCPSmSMfmZjOKLX4zoYHHTNJQW2daVgZ2PTpkASFlSA"]],
+};
+
+const mockedDnsResolveTxt = (error, addresses) => (lookup, callback) => callback(error, addresses);
+
+describe("Resolver", () => {
+  let resolver;
+
+  const resolve = () => resolver.resolve("dummy-domain.com");
+
+  beforeEach(() => {
+    resolver = new Resolver();
+  });
+
+  describe(".validateRequest", () => {
+    it("throws on invalid domain", () => {
+      expect(() => resolver.validateRequest({ params: { name: "xyz" } })).toThrow(InvalidRequestError);
```

Comment:
Is there a way to run these tests against an array of test cases?  For valid and invalid test cases it is good to have a range of values to test against. 

Maybe something like

invalid
 - xyz
 - xyz.com.notvalid
 - .com
 - my.weird.domain.com.very.werid

valid
 - a.com
 - xyz.com
 - sub.domain.com
 - xyz.net